core/src/main/java/org/apache/calcite/sql/SqlWithItem.java

```diff
@@ -30,13 +30,26 @@
 public class SqlWithItem extends SqlCall {
   public SqlIdentifier name;
   public @Nullable SqlNodeList columnList; // may be null
+  public SqlLiteral recursive;
```

Comment:
Do you have to use the `SqlLiteral` type? I think using `boolean` is simpler.

Reply:
SqlLiteral contains a position, so you can use it in error messages if necessary.